src/ipa_pc/mod.rs

```diff
@@ -885,91 +963,105 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
-                    })?;
+        for (label, (_point_label, point)) in query_set.iter() {
+
+            let labeled_polynomial =
+                poly_map.get(label).ok_or(Error::MissingPolynomial {
+                    label: label.to_string(),
+                })?;
 
-                points.push(point);
+            points.push(point);
 
-                // y_i
-                let evaluated_y = labeled_polynomial.polynomial().evaluate(point); 
+            // y_i
+            let evaluated_y = labeled_polynomial.polynomial().evaluate(*point);
 
-                // (p_i(X) - y_i) / (X - x_i)
-                let polynomial = 
-                    &(labeled_polynomial.polynomial() - &Polynomial::from_coefficients_vec(vec![evaluated_y])) 
-                    / 
+            // (p_i(X) - y_i) / (X - x_i)
+            let polynomial =
+                &(labeled_polynomial.polynomial() - &Polynomial::from_coefficients_vec(vec![evaluated_y]))
+                    /
                     &Polynomial::from_coefficients_vec(vec![
-                        (G::ScalarField::zero() - point), 
+                        (G::ScalarField::zero() - point),
                         G::ScalarField::one()
                     ]);
-                
-                // h(X) = SUM( lambda^i * ((p_i(X) - y_i) / (X - x_i)) )
-                batch_polynomial += (cur_challenge, &polynomial);
 
-                // lambda^i
-                cur_challenge = opening_challenges(opening_challenge_counter);
-                opening_challenge_counter += 1;
-            }
+            // h(X) = SUM( lambda^i * ((p_i(X) - y_i) / (X - x_i)) )
+            batch_polynomial += (cur_challenge, &polynomial);
+
+            // lambda^i
+            cur_challenge = opening_challenges(opening_challenge_counter);
+            opening_challenge_counter += 1;
         }
 
+        let key_len = ck.comm_key.len();
+        let p_len = batch_polynomial.coeffs.len();
+
         // Commitment of the h(X) polynomial
-        let batch_commitment = Self::cm_commit(
-            ck.comm_key.as_slice(),
-            batch_polynomial.coeffs.as_slice(),
-            None,
-            None,
-        ).into_affine();
+        let batch_commitment: Vec<G>;
+
+        if p_len > key_len {
+
+            batch_commitment = (0..p_len / key_len + if p_len % key_len != 0 { 1 } else { 0 }).into_par_iter().map(
+                |i| {
+                    Self::cm_commit(
+                        &ck.comm_key,
+                        &batch_polynomial.coeffs[i * key_len..core::cmp::min((i + 1) * key_len, p_len)],
+                        None,
+                        None,
+                    ).into_affine()
+                }
+            ).collect();
+
+        } else {
+
+            batch_commitment = vec![
+                Self::cm_commit(
+                    ck.comm_key.as_slice(),
+                    batch_polynomial.coeffs.as_slice(),
+                    None,
```

Comment:
Also here: hiding randomness if one of the involved polys need to be secured.

Reply:
Done